src/policy.c

```diff
@@ -1083,15 +1119,21 @@ void qd_policy_amqp_open(qd_connection_t *qd_conn) {
         }
 
         if (qd_policy_open_lookup_user(policy, qd_conn->user_id, hostip, vhost, conn_name,
-                                       settings_name, SETTINGS_NAME_SIZE, conn_id,
-                                       qd_conn->policy_settings) &&
+                                       settings_name, SETTINGS_NAME_SIZE, conn_id) &&
             settings_name[0]) {
             // This connection is allowed by policy.
             // Apply transport policy settings
-            if (qd_conn->policy_settings->maxFrameSize > 0)
-                pn_transport_set_max_frame(pn_trans, qd_conn->policy_settings->maxFrameSize);
-            if (qd_conn->policy_settings->maxSessions > 0)
-                pn_transport_set_channel_max(pn_trans, qd_conn->policy_settings->maxSessions - 1);
+            if (qd_policy_open_fetch_settings(policy, qd_conn->user_id, hostip, vhost, conn_name,
+                                        settings_name, conn_id,
+                                        qd_conn->policy_settings)) {
+                if (qd_conn->policy_settings->maxFrameSize > 0)
+                    pn_transport_set_max_frame(pn_trans, qd_conn->policy_settings->maxFrameSize);
+                if (qd_conn->policy_settings->maxSessions > 0)
+                    pn_transport_set_channel_max(pn_trans, qd_conn->policy_settings->maxSessions - 1);
+            } else {
+                // failed to fetch settings
+                connection_allowed = false;
```

Comment:
so if there was some issue fetching the policy settings from the python side, the connection will be closed ? Is there a way we can try to pre load the policy settings and warn if that fails? Please ignore this comment if this is already happening.

Reply:
The common pattern in dispatch C code is if any qd_entity_get call fails then the entity in question is failed. Same goes for connector connection policy: if the policy does not load then that connection is denied. Pre-loading the policy is a separate issue. Suppose there was a preload that failed then all the connections using that failed preload must fail as well.